patroni/ctl.py

```diff
@@ -1404,6 +1416,151 @@ def _do_failover_or_switchover(action: str, cluster_name: str, group: Optional[i
+        else:
+            group = click.prompt('Citus group', type=int)
+            dcs = get_dcs(cluster_name, group)
+            cluster = dcs.get_cluster()
+
+    config = global_config.from_cluster(cluster)
+
+    cluster_leader = cluster.leader and cluster.leader.name
+
+    if not cluster_leader:
+        raise PatroniCtlException('This cluster has no leader')
+
+    if cluster.leader and cluster.leader.multisite:
+        leader_site = (cluster.leader.multisite.get('name') if not cluster.leader.multisite.get('standby_config') else
+                       cluster.leader.multisite.get('standby_config', {}).get('leader_site'))
+    else:
+        raise PatroniCtlException('Multisite is not active or there is no leader site, cannot switch sites')
+
+    if switchover_leader is None:
+        if force:
+            switchover_leader = leader_site
+        else:
+            prompt = 'Leader site'
+            switchover_leader = click.prompt(prompt, type=str, default=leader_site)
+
+    if leader_site != switchover_leader:
+        raise PatroniCtlException(f'Site {switchover_leader} is not the leader of cluster {cluster_name}')
+
+    # multisite_dcs = get_dcs(cluster_name, group, True)
+    # multisite_cluster = multisite_dcs.get_cluster()
+
+    candidate_names = [str(m.multisite['name']) for m in cluster.members
+                       if m.multisite and m.multisite['name'] != leader_site]
+    # We sort the names for consistent output to the client
+    candidate_names.sort()
+
+    # TODO: once there is a reliable way for getting the candidate sites when on the leader site, turn this back on
+    # if not candidate_names:
+    #     raise PatroniCtlException('No candidates found to switch over to')
+
+    if candidate is None and not force:
+        candidate = click.prompt('Candidate ' + str(candidate_names), type=str, default='')
+
+    if candidate and candidate not in candidate_names:
+        if candidate == cluster_leader:
+            raise PatroniCtlException(
+                f'Site {candidate} is already the leader of cluster {cluster_name}')
+        raise PatroniCtlException(
+            f'Site {candidate} does not exist in cluster {cluster_name}')
+
+    scheduled_at_str = None
+    scheduled_at = None
+
+    if scheduled is None and not force:
+        next_hour = (datetime.datetime.now() + datetime.timedelta(hours=1)).strftime('%Y-%m-%dT%H:%M')
+        scheduled = click.prompt('When should the switchover take place (e.g. ' + next_hour + ' ) ',
+                                 type=str, default='now')
+
+        scheduled_at = parse_scheduled(scheduled)
+        if scheduled_at:
+            if config.is_paused:
+                raise PatroniCtlException("Can't schedule switchover in the paused state")
+            scheduled_at_str = scheduled_at.isoformat()
+
+    switchover_value = {'candidate': candidate}
+
+    if scheduled_at_str:
+        switchover_value['scheduled_at'] = scheduled_at_str
```

Comment:
This value is ignored by the multisite code. Scheduled  switchover is not yet implemented.